cinn/auto_schedule/task/task_optimizer.h

```diff
@@ -31,7 +32,7 @@ namespace auto_schedule {
 // optimal schedule for the task.
 class TaskOptimizer {
  public:
-  TaskOptimizer(const TuneTask& task, ScheduleMeasurer* schedule_measurer);
+  TaskOptimizer(const TuneTask& task, ScheduleMeasurer* schedule_measurer, std::shared_ptr<Database> database);
```

Comment:
TaskOptimizer不持有Database，生命期不会长于Database，不适用shared_ptr场景
```suggestion
  TaskOptimizer(const TuneTask& task, ScheduleMeasurer* schedule_measurer, Database* database);
```

Reply:
done